### PR TITLE
Use audienceActions to determine prompt type

### PR DESCRIPTION
**Pull request by kristenwang**
b/270965538. Currently we use the experiment flag to determine whether to show survey first or contribution prompt first, and we want to deprecated the flag in the future and use the order of actions from audienceActions to determine the prompt to show here.

### CODE REVIEW

**Review by ChrisAntaki — APPROVED** [review]

**Comment by kristenwang** [comment]
After further discussion with Ricky, I changed the logic because we want the following to happen:
1. If the SURVEY_TRIGGERING_PRIORITY experiment is turned on, we will show survey prompt first if it's not previously dismissed.
2. If SURVEY_TRIGGERING_PRIORITY experiment is turned off, and we have contribution in the audience actions, then we will show the first prompt (not dismissed before) from audience actions.
3. If SURVEY_TRIGGERING_PRIORITY experiment is turned off, and we don't have the contribution in the audience actions, we will follow the current behavior.

File: src/runtime/auto-prompt-manager.js [hunk 1]
```diff
@@ -502,37 +502,40 @@ export class AutoPromptManager {
...
         ExperimentFlags.SURVEY_TRIGGERING_PRIORITY
       );
 
-      // If audienceActions contains both contribution and survey actions,
-      // we will use the order from audienceActions to determine the preference.
-      const contributionIndex = potentialActions.findIndex(
+      let previousPrompts = [];
+      if (dismissedPrompts) {
+        previousPrompts = dismissedPrompts.split(',');
+        previousPrompts = previousPrompts.map((action) =>
```
Comment (ChrisAntaki) [inline src/runtime/auto-prompt-manager.js:514]: Could we combine this line with the line above?
```suggestion
         previousPrompts.map((action) =>
```

Comment (ChrisAntaki) [reply src/runtime/auto-prompt-manager.js:514]: Ex: `previousPrompts = dismissedPrompts.split(',').map(...`

File: src/runtime/auto-prompt-manager.js [hunk 2]
```diff
@@ -502,26 +502,40 @@ export class AutoPromptManager {
...
-        await this.isExperimentEnabled_(
-          article,
-          ExperimentFlags.SURVEY_TRIGGERING_PRIORITY
+      // Survery take highest priority if this flag is enabled.
+      const prioritizeSurvey = await this.isExperimentEnabled_(
+        article,
+        ExperimentFlags.SURVEY_TRIGGERING_PRIORITY
+      );
+
+      let previousPrompts = [];
```
Comment (ChrisAntaki) [inline src/runtime/auto-prompt-manager.js:511]: Nit: This name could be clearer
```suggestion
      let previouslyShownPrompts = [];
```

File: src/runtime/auto-prompt-manager.js [hunk 3]
```diff
@@ -502,26 +502,40 @@ export class AutoPromptManager {
       autoPromptType === AutoPromptType.CONTRIBUTION ||
       autoPromptType === AutoPromptType.CONTRIBUTION_LARGE
     ) {
-      const preferSurveyOverContributionPrompt =
-        await this.isExperimentEnabled_(
-          article,
-          ExperimentFlags.SURVEY_TRIGGERING_PRIORITY
+      // Survery take highest priority if this flag is enabled.
+      const prioritizeSurvey = await this.isExperimentEnabled_(
```
Comment (ChrisAntaki) [inline src/runtime/auto-prompt-manager.js:506]: We don't need this variable until later on in the code. Could you move the definition closer to where it's first used?

File: src/runtime/auto-prompt-manager.js (hunk 1 above)
Comment (kristenwang) [reply src/runtime/auto-prompt-manager.js]: Done

File: src/runtime/auto-prompt-manager.js (hunk 3 above)
Comment (kristenwang) [reply src/runtime/auto-prompt-manager.js]: Done.

**Comment by ChrisAntaki** [comment]
Looks good 👍

File: src/runtime/auto-prompt-manager.js (hunk 2 above)
Comment (kristenwang) [reply src/runtime/auto-prompt-manager.js]: Done.

**Pull request merged**